src/poc/miner_poc_grpc_client_statem.erl

```diff
@@ -0,0 +1,1080 @@
+                Res
+        end
+    catch
+        _Class:_Error:_Stack ->
+            lager:warning(
+                "error whilst connectting to validator, will try again in a bit. Reason: ~p, Details: ~p, Stack: ~p",
+                [_Class, _Error, _Stack]
+            ),
+            {error, connect_validator_failed}
+    end.
+
+-spec connect_stream_poc(
+    Connection::grpc_client_custom:connection(),
+    SelfPubKeyBin::libp2p_crypto:pubkey_bin(),
+    SelfSigFun::function()) ->
+        {error, any()} | {ok, pid()}.
+connect_stream_poc(Connection, SelfPubKeyBin, SelfSigFun) ->
+    case miner_poc_grpc_client_handler:poc_stream(Connection, SelfPubKeyBin, SelfSigFun) of
+        {error, _Reason} = Error ->
+            lager:debug("failed to connect to poc stream on connection ~p, reason: ~p", [Connection, _Reason]),
+            Error;
+        {ok, Stream} = Res ->
+            lager:debug("successfully connected poc stream ~p on connection ~p", [Stream, Connection]),
+            Res
+    end.
+
+-spec connect_stream_config_update(
+    Connection::grpc_client_custom:connection()) ->
+        {error, any()} | {ok, pid()}.
+connect_stream_config_update(Connection) ->
+   case miner_poc_grpc_client_handler:config_update_stream(Connection) of
+        {error, _Reason} = Error ->
+            lager:debug("failed to connect to config stream on connection ~p, reason: ~p", [Connection, _Reason]),
+            Error;
+        {ok, Stream} = Res ->
+            lager:debug("successfully connected config update stream ~p on connection ~p", [
+                Stream, Connection
+            ]),
+            Res
+    end.
+
+-spec connect_stream_region_params_update(
+    Connection::grpc_client_custom:connection(),
+    SelfPubKeyBin::libp2p_crypto:pubkey_bin(),
+    SelfSigFun::function()
+) -> {error, any()} | {ok, pid()}.
+connect_stream_region_params_update(Connection, SelfPubKeyBin, SelfSigFun) ->
+    case
+        miner_poc_grpc_client_handler:region_params_update_stream(
+            Connection, SelfPubKeyBin, SelfSigFun
+        )
+    of
+        {error, _Reason} = Error ->
+            lager:debug("failed to connect to region params update stream on connection ~p, reason: ~p", [Connection, _Reason]),
+            Error;
+        {ok, Stream} = Res ->
+            lager:debug(
+                "successfully connected region params update stream ~p on connection ~p",
+                [Stream, Connection]
+            ),
+            Res
+    end.
+
+-spec send_report(
+    ReportType::witness | receipt,
+    Report::#blockchain_poc_receipt_v1_pb{} | #blockchain_poc_witness_v1_pb{},
+    OnionKeyHash::binary(),
+    SelfPubKeyBin::libp2p_crypto:pubkey_bin(),
+    SigFun::function(),
+    Connection::grpc_client_custom:connection(),
+    RetryAttempts::non_neg_integer()
+) -> ok.
+send_report(_ReportType, _Report, _OnionKeyHash, _SelfPubKeyBin, _SigFun, _Connection, 0) ->
+    ok;
```

Comment:
Log here?

Reply:
not sure how much value there is in logging this but ok

Reply:
https://github.com/helium/miner/pull/1068/commits/32247902ed59dcdcf354746ce707c1a1aa4c5eda